command/agent/auth/token-file/token_file.go

```diff
@@ -0,0 +1,119 @@
+package token_file
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/command/agent/auth"
+)
+
+type TokenFileMethod struct {
+	logger    hclog.Logger
+	mountPath string
+
+	cachedToken                 string
+	tokenFilePath               string
+	followSymlinks              bool
+	removeTokenFileAfterReading bool
+}
+
+func NewTokenFileAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
+	if conf == nil {
+		return nil, errors.New("empty config")
+	}
+	if conf.Config == nil {
+		return nil, errors.New("empty config data")
+	}
+
+	a := &TokenFileMethod{
+		logger:    conf.Logger,
+		mountPath: "auth/token",
+	}
+
+	tokenFilePathRaw, ok := conf.Config["token_file_path"]
```

Comment:
Note that we error if this isn't provided - I decided during implementation that this felt better than defaulting to `~/.vault-token`, as none of our other auth methods have defaults for file paths, really.

Reply:
Cool, while that might've been convenient, I prefer it be explicit.  Less surprises that way.